custom_components/upnp_availability/upnpstatustracker.py

```diff
@@ -194,16 +195,26 @@ async def find_devices(self, async_callback=None):
             async_callback = self.handle_alive
 
         for source_address in self.source_addresses:
+            _LOGGER.debug("Starting device search using addr %s", source_address)
             try:
-                await async_search(
-                    service_type=ROOT_DEVICE,
-                    source_ip=ip_address(source_address),
-                    async_callback=async_callback,
-                )
-            except OSError as ex:
-                _LOGGER.warning(
-                    "Unable to search using addr %s: %s", source_address, ex
-                )
+                if type(ip_address(source_address)) is IPv6Address:
+                    source_ipv6 = source_address.split("%")
+                    source_address = (source_ipv6[0],0,0,source_ipv6[1])
+                    return source_address
+                elif type(ip_address(source_address)) is IPv4Address:
+                    source_address=(source_address,0)
+                try:
+                    await async_search(
+                        service_type=ROOT_DEVICE,
+                        source=source_address,
+                        async_callback=async_callback,
+                    )
+                except OSError as ex:
+                    _LOGGER.warning(
+                        "Unable to search using addr %s: %s", source_address, ex
+                    )
```

Comment:
Move this to a new internal `_search` that takes the formatted source address. That will make part much cleaner and easier to follow.